http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/sender/websocket/WebSocketClientConnector.java

```diff
@@ -0,0 +1,184 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+package org.wso2.carbon.transport.http.netty.sender.websocket;
+
+import org.wso2.carbon.messaging.BinaryCarbonMessage;
+import org.wso2.carbon.messaging.CarbonCallback;
+import org.wso2.carbon.messaging.CarbonMessage;
+import org.wso2.carbon.messaging.CarbonMessageProcessor;
+import org.wso2.carbon.messaging.ClientConnector;
+import org.wso2.carbon.messaging.ControlCarbonMessage;
+import org.wso2.carbon.messaging.Headers;
+import org.wso2.carbon.messaging.TextCarbonMessage;
+import org.wso2.carbon.messaging.exceptions.ClientConnectorException;
+import org.wso2.carbon.transport.http.netty.common.Constants;
+import org.wso2.carbon.transport.http.netty.internal.HTTPTransportContextHolder;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.net.ssl.SSLException;
+
+/**
+ * WebSocket Client connector which connects to a remote WebSocket Endpoint.This store the WebSocket Clients
+ * against a unique ID given by the user. So make sure to add a unique ID the carbon message in order to use
+ * this properly.
+ */
+public class WebSocketClientConnector implements ClientConnector {
+
+    // Map<uniqueID, webSocketClient>
+    private final Map<String, WebSocketClient> webSocketClientMap = new ConcurrentHashMap<>();
+
+    @Override
+    public boolean send(CarbonMessage msg, CarbonCallback callback) throws ClientConnectorException {
+        if (msg instanceof TextCarbonMessage) {
+            TextCarbonMessage textCarbonMessage = (TextCarbonMessage) msg;
+            sendText(textCarbonMessage);
+        } else if (msg instanceof ControlCarbonMessage) {
+            ControlCarbonMessage controlCarbonMessage = (ControlCarbonMessage) msg;
+            handleControlMessage(controlCarbonMessage);
+        } else if (msg instanceof BinaryCarbonMessage) {
+            BinaryCarbonMessage binaryCarbonMessage = (BinaryCarbonMessage) msg;
+            sendBinary(binaryCarbonMessage);
+        }
+        return true;
```

Comment:
The client connector is a API which coming under connector framework. So this method is overridden. If the message sending is failed then this returns fales. But we manage if for now by other ways for WebSocket. I will look into this. Thanks.